@cfriedt any chance you could also look into the other issue please?

```
INFO    - 1) sample.net.lwm2m_client.data_cache on qemu_x86/atom error (Build failure - undefined reference to `gettimeofday')
```

Setting `CONFIG_XSI_SINGLE_PROCESS=y` in `samples/net/lwm2m_client/overlay-data-cache.conf` does the trick but it doesn't sound right since AFAICS the LWM2M data cache feature doesn't really use any of the functions from this option group. 
Thanks!

--- a/lib/posix/options/clock_common.c
+++ b/lib/posix/options/clock_common.c
@@ -59,12 +59,10 @@ int z_vrfy___posix_clock_get_base(clockid_t clock_id, struct timespec *ts)
 
 int z_clock_gettime(clockid_t clock_id, struct timespec *ts)
 {
-	struct timespec base;
+	struct timespec base = {.tv_sec = 0, .tv_nsec = 0};
 
 	switch (clock_id) {
 	case CLOCK_MONOTONIC:
-		base.tv_sec = 0;
-		base.tv_nsec = 0;
 		break;
 
 	case CLOCK_REALTIME:

